Packs/AggregatedScripts/Scripts/QuarantineFile/QuarantineFile.py

```diff
@@ -0,0 +1,1144 @@
+                        brand="Unknown",
+                        script_args=self.script_args,
+                    )
+                )
+
+        demisto.debug(f"[EndpointBrandMapper] Processing complete. Found {len(online_endpoints)} online endpoints.")
+        return online_endpoints
+
+
+""" BRAND HANDLER INTERFACE & FACTORY """
+
+
+class BrandHandler(ABC):
+    """Abstract base class (Interface) for all brand-specific handlers."""
+
+    def __init__(self, brand: str, orchestrator):
+        """
+        Initializes the BrandHandler.
+
+        Args:
+            brand (str): The name of the brand this handler is responsible for.
+            orchestrator (QuarantineOrchestrator): The orchestrator instance.
+        """
+        self.brand = brand
+        self.orchestrator = orchestrator
+
+    @abstractmethod
+    def validate_args(self, args: dict) -> None:
+        """
+        Validates that all required arguments for this brand are present.
+        """
+
+    @abstractmethod
+    def run_pre_checks_and_get_initial_results(self, args: dict) -> tuple[list, list[QuarantineResult]]:
+        """
+        Runs brand-specific pre-checks, like checking if a file is already quarantined.
+        """
+
+    @abstractmethod
+    def initiate_quarantine(self, args: dict) -> dict:
+        """
+        Initiates the quarantine action for the brand and returns a polling job object.
+        """
+
+    @abstractmethod
+    def finalize(self, job: dict, last_poll_response: list) -> list[QuarantineResult]:
+        """
+        Processes the final results of a completed polling job for the brand.
+        """
+
+
+class XDRHandler(BrandHandler):
+    """Concrete handler for Cortex XDR and Cortex Core quarantine actions."""
+
+    CORE_COMMAND_PREFIX = "core"
+    XDR_COMMAND_PREFIX = "xdr"
+
+    def __init__(self, brand: str, orchestrator):
+        """
+        Initializes the XDRHandler.
+
+        Args:
+            brand (str): The brand name ('Cortex XDR - IR' or 'Cortex Core - IR').
+            orchestrator (QuarantineOrchestrator): The main orchestrator instance.
+        """
+        super().__init__(brand, orchestrator)
+        self.command_prefix = self.CORE_COMMAND_PREFIX if self.brand == Brands.CORTEX_CORE_IR else self.XDR_COMMAND_PREFIX
+
+    def validate_args(self, args: dict) -> None:
+        """
+        Validates that the 'file_path' argument is provided for XDR actions.
+
+        Args:
+            args (dict): The script arguments.
+
+        Raises:
+            ValueError: If the 'file_path' argument is missing.
+        """
+        demisto.debug(f"[{self.brand} Handler] Validating args.")
+        if not args.get(QuarantineOrchestrator.FILE_PATH_ARG):
+            raise DemistoException(f"The '{QuarantineOrchestrator.FILE_PATH_ARG}' argument is required for brand {self.brand}.")
+
+    def _endpoint_already_quarantined(self, endpoint_id: str, file_hash: str, file_path: str) -> bool:
```

Comment:
If I got your correctly this is the function that polls for the status to see if it's done, not sure "_endpoint_already_quarantined" is the right name